### PR TITLE
Updating pushy client to use the multiple gemspec pattern

### PR DESCRIPTION
**Pull request by tyler-ball**
As pioneered by @jkeiser. This allow the ChefDK to pull in both regular and windows specific dependencies. Without these changes we have to [duplicate](https://github.com/chef/chef-dk/blob/13d4becd249f03a3e63d821b6facfe9182528584/Gemfile#L97-L110) the pushy Windows specific dependencies. 

\cc @chef/server-team

### CODE REVIEW

**Comment by tyler-ball** [comment]
Also \cc @jkeiser - Am I doing this right?

File: Gemfile [hunk 1]
```diff
@@ -1,24 +1,16 @@
...
-  gem "win32-service"
-end
-
-platforms :mingw_18 do
-  gem "win32-open3"
+# Pick the gemspec for our platform
+gemspec_name = "opscode-pushy-client"
+Dir.glob("#{gemspec_name}-*.gemspec").each do |gemspec_filename|
+  gemspec_filename =~ /^#{gemspec_name}-(.+).gemspec/
+  gemspec_platform = $1
```
Comment (afiune) [inline Gemfile]: Where is `$1` coming from?

**Comment by afiune** [comment]
Just a little question/doubt but overall I'm 👍

File: Gemfile (hunk 1 above)
Comment (tyler-ball) [inline Gemfile]: The regex on the line before. It matches the `(.+)` matcher

**Comment by tyler-ball** [comment]
Okay, I'm going to close this. It turns out that we are already getting the correct dependencies in the ChefDK because the Chef gem is 'doing the right thing'.

Chef has 2 gemspecs - 1 for all systems and 1 that adds additional dependencies on Windows. John added all the crazy logic that makes that work and [defines](https://github.com/chef/chef/blob/0e52b49d30849472a9ad829eaee488536ee540cb/Gemfile.lock#L48) 2 specs for Chef. Because Chef is a dependency of Pushy when we `bundle exec rake install` pushy on a Windows system we automatically get all the windows specific dependencies (like `win32-process`).

The right fix in the long term is to do the same thing for pushy. It should define 2 specs and the windows specific spec would have all the `win32` dependencies listed in it. But doing that today requires copying around a massive amount of code (the custom rake tasks). Even after doing that, `bundle exec rake install` will still fail because there are then 2 gemspecs. So more custom rake tasks need to be written which know how to build the Pushy gem where there are 2 gemspecs present.

Because nothing is broken today (both the ChefDK and Pushy omnibus packages have the necessary `win32` dependencies) and there isn't good support for doing the right fix, I'm going to close this. Once the Rake tasks that know how to do all this custom packaging are extracted into an external gem, we could come back and apply those fixes here.

**Pull request closed without merging**